src/components/edit/EditBarbell.tsx

```diff
@@ -55,7 +55,7 @@ const useEditBarbellAPI = (props: EditBarbellProps) => {
   const handleAdd = React.useCallback(
     (increment: number) => {
       setTargetWeightHistory((prev) => prev.push(targetWeight));
-      onTargetWeightChange(targetWeight + increment * 2);
+      onTargetWeightChange((prevWeight) => prevWeight + increment * 2);
```

Comment:
here is the actual fix. By using prevWeight, it batches updates, but each one is a function that can run _after_ the change, vs jus statically setting it before the other ones can "finish"